src/course-unit/course-sequence/sequence-navigation/SequenceNavigationTabs.jsx

```diff
@@ -1,23 +1,48 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
 import { Button } from '@edx/paragon';
 import { Plus as PlusIcon } from '@edx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
+import { useNavigate } from 'react-router-dom';
 
-import { useIndexOfLastVisibleChild } from '../hooks';
+import { addNewSequenceNavigationUnit } from '../../data/thunk';
+import { addNewUnitId, changeTitleEditFormOpen, updateQueryPendingStatus } from '../../data/slice';
+import { getCourseId, getNewUnitId, getSequenceId } from '../../data/selectors';
 import messages from '../messages';
+import { useIndexOfLastVisibleChild } from '../hooks';
 import SequenceNavigationDropdown from './SequenceNavigationDropdown';
 import UnitButton from './UnitButton';
 
 const SequenceNavigationTabs = ({ unitIds, unitId }) => {
   const intl = useIntl();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const sequenceId = useSelector(getSequenceId);
+  const newUnitId = useSelector(getNewUnitId);
+  const courseId = useSelector(getCourseId);
+
   const [
     indexOfLastVisibleChild,
     containerRef,
     invisibleStyle,
   ] = useIndexOfLastVisibleChild();
   const shouldDisplayDropdown = indexOfLastVisibleChild === -1;
 
+  const handleAddNewSequenceUnit = () => {
+    dispatch(addNewSequenceNavigationUnit(unitId, sequenceId));
+    dispatch(updateQueryPendingStatus(true));
+  };
+
+  useEffect(() => {
+    if (newUnitId) {
+      const pathToNewSequenceUnit = `/course/${courseId}/container/${newUnitId}/${sequenceId}`;
+      navigate(pathToNewSequenceUnit, { replace: true });
+      dispatch(addNewUnitId(''));
```

Comment:
```suggestion
      dispatch(addNewUnitId(null));
```
Super nit: Like in the initial state

Reply:
Removed